crates/ruma-state-res/tests/it/main.rs

```diff
@@ -0,0 +1,249 @@
+struct Pdu {
+    event_id: OwnedEventId,
+    room_id: OwnedRoomId,
+    sender: OwnedUserId,
+    origin_server_ts: MilliSecondsSinceUnixEpoch,
+    #[serde(rename = "type")]
+    kind: TimelineEventType,
+    content: Box<RawJsonValue>,
+    state_key: Option<String>,
+    prev_events: Vec<OwnedEventId>,
+    auth_events: Vec<OwnedEventId>,
+    redacts: Option<OwnedEventId>,
+}
+
+impl Event for Pdu {
+    type Id = OwnedEventId;
+
+    fn event_id(&self) -> &Self::Id {
+        &self.event_id
+    }
+
+    fn room_id(&self) -> &RoomId {
+        &self.room_id
+    }
+
+    fn sender(&self) -> &UserId {
+        &self.sender
+    }
+
+    fn origin_server_ts(&self) -> MilliSecondsSinceUnixEpoch {
+        self.origin_server_ts
+    }
+
+    fn event_type(&self) -> &TimelineEventType {
+        &self.kind
+    }
+
+    fn content(&self) -> &RawJsonValue {
+        &self.content
+    }
+
+    fn state_key(&self) -> Option<&str> {
+        self.state_key.as_deref()
+    }
+
+    fn prev_events(&self) -> Box<dyn DoubleEndedIterator<Item = &Self::Id> + '_> {
+        Box::new(self.prev_events.iter())
+    }
+
+    fn auth_events(&self) -> Box<dyn DoubleEndedIterator<Item = &Self::Id> + '_> {
+        Box::new(self.auth_events.iter())
+    }
+
+    fn redacts(&self) -> Option<&Self::Id> {
+        self.redacts.as_ref()
+    }
+}
+
+/// Extract `.content.room_version` from a PDU
+#[derive(Deserialize)]
+struct ExtractRoomVersion {
+    room_version: RoomVersionId,
+}
+
+/// Type describing a resolved state event
+#[derive(Serialize)]
+struct ResolvedState {
+    kind: StateEventType,
+    state_key: String,
+    event_id: OwnedEventId,
+    content: JsonValue,
+}
+
+impl PartialEq for ResolvedState {
+    fn eq(&self, other: &Self) -> bool {
+        self.kind == other.kind
+            && self.state_key == other.state_key
+            && self.event_id == other.event_id
+    }
+}
+
+impl Eq for ResolvedState {}
+
+impl Ord for ResolvedState {
+    fn cmp(&self, other: &Self) -> Ordering {
+        Ordering::Equal
+            .then(self.kind.cmp(&other.kind))
+            .then(self.state_key.cmp(&other.state_key))
+            .then(self.event_id.cmp(&other.event_id))
+    }
+}
+
+impl PartialOrd for ResolvedState {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+/// Results of a test
+struct TestResults {
+    /// The resolved state of the room
+    resolved_state: BTreeSet<ResolvedState>,
+}
+
+/// Test a JSON file containing a list of PDUs and return the results
+fn run<P>(path: P) -> Result<TestResults, Box<dyn Error>>
+where
+    P: AsRef<Path>,
+{
+    // Run `cargo test -- --show-output` to view traces, set `RUST_LOG` to control filtering
+    let _subscriber = tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .with_test_writer()
+        .finish()
+        .set_default();
+
+    let path = PathBuf::from("tests/it/fixtures").join(path);
```

Comment:
```suggestion
    let path = Path::from("tests/it/fixtures").join(path)
```

nit: this is just calling `Path::join` through the auto-deref. Surprised clippy didn't catch this.